docs/docs/building-modules/09-module-interfaces-continued.md

```diff
@@ -0,0 +1,158 @@
+
+To add a custom command or query, you can use the `Builder.AddCustomCommand` or `Builder.AddCustomQuery` methods, respectively. These methods take a `cobra.Command` or `cobra.Command` instance, respectively, which can be used to define the behavior of the command or query.
+
+### Advanced Usage
+
+#### Specifying Subcommands
+
+By default, `autocli` generates a command for each method in your gRPC service. However, you can specify subcommands to group related commands together. To specify subcommands, you can use the `autocliv1.ServiceCommandDescriptor` struct.
+
+For example, suppose you have a service with two methods: `GetTest` and `GetTest2`. You can group these methods together under a `get` subcommand using the following code
+
+```
+autoCliOpts := &autocliv1.ModuleOptions{
+    Tx: &autocliv1.ServiceCommandDescriptor{
+        Service: myServiceDesc.ServiceName,
+        SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+            "get": {
+                Service: myServiceDesc.ServiceName,
+                SubCommands: map[string]*autocliv1.ServiceCommandDescriptor{
+                    "test": {Service: myServiceDesc.ServiceName, Method: "GetTest"},
+                    "test2": {Service: myServiceDesc.ServiceName, Method: "GetTest2"},
+                },
+            },
+        },
+    },
+}
+
+
+```
+
+With this configuration, you can invoke the `GetTest` method by running `./app tx myservice get test`, and you can invoke the `GetTest2` method by running `./app tx myservice get test2`.
+
+#### Adding Custom Commands
+
+Although `autocli` can automatically generate CLI commands and flags based on your Protocol Buffer messages, you may want to add custom commands to your CLI application. To add a custom command, you can use the `AddCommand()` method provided by `cobra`. For instance, to add a custom `test` command to your CLI application, you can use the following code:
+
+```
+rootCmd := autocli.RootCmd(appOptions)
+rootCmd.AddCommand(&cobra.Command{
+    Use:   "test",
+    Short: "custom test command",
+    RunE: func(cmd *cobra.Command, args []string) error {
+        // custom command logic here
+        return nil
+    },
+})
+
+```
+
+This will add a new `test` command to your CLI application that can be invoked by running `./app test`.
+
+### Customising Flag Names
+
+By default, `autocli` generates flag names based on the names of the fields in your Protocol Buffer message. However, you can customise the flag names by providing a `namingOptions` parameter to the `Builder.AddMessageFlags()` method. This parameter allows you to specify custom names for flags based on the names of the message fields. For example, if you have a message with the fields `test` and `test1`, you can use the following naming options to customise the flags
+```
+options := autocliv1.RpcCommandOptions{ 
```

Comment:
This is not clear where it should be, we should precise it should be defined in the module auto cli options and that it is a field of the `ServiceCommandDescriptor` struct.

Reply:
Ive added a few more notes as to where everything should live, also realised I made an error in the naming. Re the snippet, the only thing I think we could reference (unless you have other ideas) is define an instance of `autocliv1.ModuleOptions` which we have already referenced in subcommands but could be referenced again if needed.